app/Check/Consumers/Client/ClientBuilder.php

```diff
@@ -0,0 +1,90 @@
+<?php declare(strict_types = 1);
+
+namespace Pd\Monitoring\Check\Consumers\Client;
+
+class ClientBuilder
+{
+
+	private ?int $connectTimeout;
+
+	private ?int $timeout;
+
+	/**
+	 * @see https://docs.guzzlephp.org/en/stable/request-options.html#verify
+	 */
+	private bool $verify = TRUE;
+
+	private ?\Pd\Monitoring\Check\Consumers\Client\Conf\AllowRedirectsInterface $allowRedirects = NULL;
+
+
+	private function __construct(
+		?int $connectTimeout,
+		?int $timeout
+	)
+	{
+		$this->connectTimeout = $connectTimeout;
+		$this->timeout = $timeout;
+	}
+
+
+	public function withVerify(bool $verify): self
+	{
+		$factory = clone $this;
+		$factory->verify = $verify;
+
+		return $factory;
+	}
+
+
+	public function withAllowRedirects(\Pd\Monitoring\Check\Consumers\Client\Conf\AllowRedirectsInterface $allowRedirects): self
+	{
+		$factory = clone $this;
+		$factory->allowRedirects = $allowRedirects;
+
+		return $factory;
+	}
+
+
+	public static function create(
+		int $connectTimeout,
+		int $timeout
+	): self
+	{
+		return new self($connectTimeout, $timeout);
+	}
+
+
+	/**
+	 * @return array<string, string|array<string, string>>
+	 */
+	private function config(): array
```

Comment:
Podle mě není důvod proč by tato metoda měla být private, je to přesně to co od toho objektu očekáváš, že bude dělat. Navíc by to podle mě měla být metoda, která ti umožní otestovat si, že ten objekt dělá co má.